mdn/kumascript.py

```diff
@@ -522,21 +523,44 @@ class CSSBox(KnownKumaScript):
     expected_scopes = set()
 
 
-class CSSxRef(KnownKumaScript):
+class XRefBase(KnownKumaScript):
+    """Base class for cross-reference KumaScript"""
+    expected_scopes = set((
+        'compatibility feature', 'specification description', 'footnote'))
+
+    def __init__(self, **kwargs):
+        super(XRefBase, self).__init__(**kwargs)
+        self.url = None
+        self.display = "(unset)"
```

Comment:
I tried this, but it means derived classes have to implement as property methods rather than simple attributes. Added some asserts that the derived classes have set these variables.